lib/jekyll/external.rb

```diff
@@ -52,34 +52,45 @@ def version_constraint(gem_name)
       #
       # names - a string gem name or array of gem names
       #
-      # rubocop:disable Metrics/MethodLength
       def require_with_graceful_fail(names)
         Array(names).each do |name|
           begin
             Jekyll.logger.debug "Requiring:", name.to_s
             require name
-          rescue LoadError
+          rescue LoadError => e
             if name.include?("-")
-              require name.tr("-", "/")
+              begin
+                require name.tr("-", "/")
```

Comment:
Why not `require_with_graceful_fail name.tr("-", "/")` and get rid of the duplication?
I guess we would still have the problem of maintaining the original `name`…

Reply:
@pathawks IMO, this route is necessary to first check if `foo-bar` can be required otherwise rescue and try loading `foo/bar` or re-raise otherwise.

Can you post your suggested Ruby code..?

Reply:
@pathawks I like the DRY version. Perhaps we can re-substitute the `/` in the name in the error-message..? Or would that be confusing for people who have configured `plugins: ['foo/bar']`